activerecord/lib/active_record/persistence.rb

```diff
@@ -55,6 +55,79 @@ def create!(attributes = nil, &block)
         end
       end
 
+      # Inserts a record (or multiple records) into the database. This method constructs a single SQL INSERT
+      # statement and sends it straight to the database. It does not instantiate the involved models and it does not
+      # trigger Active Record callbacks or validations. However, values passed to #insert_many will still go through
+      # Active Record's normal type casting and serialization.
+      #
+      # The +attributes+ parameter can be either a Hash or an Array of Hashes. These Hashes describe the
+      # attributes on the objects that are to be created.
+      #
+      # ==== Options
+      #
+      # [:returning]
+      #   (Postgres-only) An array of column names that should be returned for all successfully inserted rows.
+      #   By default, the primary key(s) will be returned for every inserted rows. Can be set to `false` to return nothing.
```

Comment:
What is the default setting of `returning`? I think adapter specific options should be listed last in the docs. Will this return `['name', 'other_column', 'yet_another_column']` when successful? or will it return `['eileen', 'whatever', 'ok'], ['other_name', 'whatever', 'ok']`?

If it's returning just the column names what's the use case for supporting this? And what's the use case for only supporting this with Postgres?

Reply:
Ah! My docs aren't clear.

Postgres supports a `RETURNING` clause on `INSERT` like this:

```sql
INSERT INTO x (name, other_column) VALUES (1, 'a'), (2, 'b') RETURNING id, name;
```

Now when the `INSERT` finishes, it'll return like a `SELECT id, name` but only for the newly inserted records.

It's handy especially to get the IDs of everything you bulk-inserted; and sometimes it's useful to ask it to return other columns.

I believe Postgres is the only DB that supports the `RETURNING` clause; and I wrote it like this:

 - If you use this with Postgres, the default is `RETURNING id` (or whatever your primary key is)
 - You can pass `returning: false` to return nothing
 - You can pass `returning: %w{ id name }` to return whatever columns you want

:A: Do you think this logic (👆) sounds good?

:B: I'll work on clarifying the docs!